### ISSUE TITLE
Variable output steps for RNNs in Keras?

### ISSUE THREAD

**Issue by johntzwei** [keras-team/keras#7738]
In the current Keras `Recurrent` API, there is no way of specifying the length of an output sequence. The number of timesteps is dependent on the number of timesteps of the input.

I'm implementing an attention LSTM for seq2seq, and the outputs are always longer than the masked inputs.

One way for me to get around that is to remove the masking for the input and use timesteps with padding for the longest sequence. Are there better ways around this? 

It seems like it would be trivial to implement if the `input_length` keyword works for the tensorflow implementation of `K.rnn`, and it might be a useful feature to have variable output steps, where losses for RNN outputs are only calculated up to the length of the true sequence.

Thank you!

**Comment by johntzwei** [comment]
So for anyone having the same question, what you need to do is to add a timesteps argument that can be passed into the recurrent class, and don't mask your input with Keras masking. Mask it with your own vocabulary word.
